src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java

```diff
@@ -144,4 +153,58 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             zipOutputStream.close();
         }
     }
+
+    @RequestMapping(value = "json/experiments/check/zip",
+            method = RequestMethod.GET,
+            produces = "application/json;charset=UTF-8")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String
+    checkMultipleExperimentsFileValid(@RequestParam(value = "accession", defaultValue = "") List<String> accessions)
+    {
+        List<Experiment> experiments = new ArrayList<>();
+        for (var accession : accessions) {
```

Comment:
Use streams and `toImmutableList` to build lists in a more declarative way. You’ll get rid of boilerplate code.

Reply:
I thought we discussed this in `download/zip` implementation if I don't misunderstand. We need to implement this way as we want to `catch` exceptions.

Reply:
Hmmm... maybe the previous logic was slightly different. With streams you can also catch exceptions, it only gets complicated when you need to throw them out of the lambda block. In this case you can do exactly the same thing without `for`.